# CMU-MOSI Multimodal SDK

# Download and Process mosi

In [1]:
! pwd

/content


In [ ]:
%env PYTHONPATH="/env/python:/content/drive/My drive/CMU-Multimodal/CMU-MultimodalSDK"
! echo $PYTHONPATH

env: PYTHONPATH="/env/python:/content/drive/My drive/CMU-Multimodal/CMU-MultimodalSDK"
"/env/python:/content/drive/My drive/CMU-Multimodal/CMU-MultimodalSDK"


In [ ]:
from google.colab import drive

drive.mount('/content/drive')

Mounted at /content/drive


In [ ]:
cd "/content/drive/My Drive/CMU-Multimodal/CMU-MultimodalSDK"

/content/drive/My Drive/CMU-Multimodal/CMU-MultimodalSDK


### word level align

In [ ]:
!pip install validators
!pip install colorama

In [ ]:
import mmsdk
from mmsdk import mmdatasdk
import numpy

In [ ]:
#A simple averaging technique. More advanced methods can be built based on intervals.
def myavg(intervals,features):
        return numpy.average(features,axis=0)

In [ ]:
#Downloading the dataset
# cmumosi_highlevel=mmdatasdk.mmdataset(mmdatasdk.cmu_mosi.highlevel,'cmumosi/')
cmumosi_highlevel=mmdatasdk.mmdataset('cmumosi/')

[2021-11-08 14:11:28.333] | Success | Computational sequence read from file cmumosi/CMU_MOSI_TimestampedWordVectors.csd ...
[2021-11-08 14:11:28.731] | Status  | Checking the integrity of the <glove_vectors> computational sequence ...
[2021-11-08 14:11:28.731] | Status  | Checking the format of the data in <glove_vectors> computational sequence ...


[2021-11-08 14:11:28.847] | Success | <glove_vectors> computational sequence data in correct format.
[2021-11-08 14:11:28.847] | Status  | Checking the format of the metadata in <glove_vectors> computational sequence ...
[2021-11-08 14:11:28.847] | Warning | <glove_vectors> computational sequence does not have all the required metadata ... continuing 


[2021-11-08 14:11:29.086] | Success | Computational sequence read from file cmumosi/CMU_MOSI_Visual_Facet_41.csd ...
[2021-11-08 14:11:29.683] | Status  | Checking the integrity of the <FACET_4.1> computational sequence ...
[2021-11-08 14:11:29.683] | Status  | Checking the format of the data in <FACET_4.1> computational sequence ...


[2021-11-08 14:11:29.823] | Success | <FACET_4.1> computational sequence data in correct format.
[2021-11-08 14:11:29.823] | Status  | Checking the format of the metadata in <FACET_4.1> computational sequence ...
[2021-11-08 14:11:29.823] | Warning | <FACET_4.1> computational sequence does not have all the required metadata ... continuing 


[2021-11-08 14:11:30.033] | Success | Computational sequence read from file cmumosi/CMU_MOSI_Visual_Facet_42.csd ...
[2021-11-08 14:11:30.424] | Status  | Checking the integrity of the <FACET_4.2> computational sequence ...
[2021-11-08 14:11:30.424] | Status  | Checking the format of the data in <FACET_4.2> computational sequence ...


[2021-11-08 14:11:30.574] | Success | <FACET_4.2> computational sequence data in correct format.
[2021-11-08 14:11:30.574] | Status  | Checking the format of the metadata in <FACET_4.2> computational sequence ...
[2021-11-08 14:11:30.575] | Warning | <FACET_4.2> computational sequence does not have all the required metadata ... continuing 


[2021-11-08 14:11:30.843] | Success | Computational sequence read from file cmumosi/CMU_MOSI_OpenSmile_EB10.csd ...
[2021-11-08 14:11:30.993] | Status  | Checking the integrity of the <OpenSmile_emobase2010> computational sequence ...
[2021-11-08 14:11:30.993] | Status  | Checking the format of the data in <OpenSmile_emobase2010> computational sequence ...


[2021-11-08 14:11:31.122] | Success | <OpenSmile_emobase2010> computational sequence data in correct format.
[2021-11-08 14:11:31.123] | Status  | Checking the format of the metadata in <OpenSmile_emobase2010> computational sequence ...
[2021-11-08 14:11:31.123] | Warning | <OpenSmile_emobase2010> computational sequence does not have all the required metadata ... continuing 


[2021-11-08 14:11:31.336] | Success | Computational sequence read from file cmumosi/CMU_MOSI_openSMILE_IS09.csd ...
[2021-11-08 14:11:35.442] | Status  | Checking the integrity of the <b'OpenSMILE'> computational sequence ...
[2021-11-08 14:11:35.443] | Status  | Checking the format of the data in <b'OpenSMILE'> computational sequence ...


[2021-11-08 14:11:35.964] | Success | <b'OpenSMILE'> computational sequence data in correct format.
[2021-11-08 14:11:35.965] | Status  | Checking the format of the metadata in <b'OpenSMILE'> computational sequence ...
[2021-11-08 14:11:35.965] | Warning | <b'OpenSMILE'> computational sequence does not have all the required metadata ... continuing 
[2021-11-08 14:11:36.311] | Success | Computational sequence read from file cmumosi/CMU_MOSI_Visual_OpenFace_1.csd ...
[2021-11-08 14:11:41.824] | Status  | Checking the integrity of the <OpenFace_1> computational sequence ...
[2021-11-08 14:11:41.827] | Status  | Checking the format of the data in <OpenFace_1> computational sequence ...


[2021-11-08 14:11:43.336] | Success | <OpenFace_1> computational sequence data in correct format.
[2021-11-08 14:11:43.336] | Status  | Checking the format of the metadata in <OpenFace_1> computational sequence ...
[2021-11-08 14:11:43.337] | Warning | <OpenFace_1> computational sequence does not have all the required metadata ... continuing 
[2021-11-08 14:11:43.548] | Success | Computational sequence read from file cmumosi/CMU_MOSI_Visual_OpenFace_2.csd ...
[2021-11-08 14:11:48.353] | Status  | Checking the integrity of the <OpenFace_2> computational sequence ...
[2021-11-08 14:11:48.355] | Status  | Checking the format of the data in <OpenFace_2> computational sequence ...


[2021-11-08 14:11:49.467] | Success | <OpenFace_2> computational sequence data in correct format.
[2021-11-08 14:11:49.467] | Status  | Checking the format of the metadata in <OpenFace_2> computational sequence ...
[2021-11-08 14:11:49.467] | Warning | <OpenFace_2> computational sequence does not have all the required metadata ... continuing 
[2021-11-08 14:11:49.691] | Success | Computational sequence read from file cmumosi/CMU_MOSI_COVAREP.csd ...
[2021-11-08 14:11:53.459] | Status  | Checking the integrity of the <COVAREP> computational sequence ...
[2021-11-08 14:11:53.461] | Status  | Checking the format of the data in <COVAREP> computational sequence ...


[2021-11-08 14:11:54.043] | Success | <COVAREP> computational sequence data in correct format.
[2021-11-08 14:11:54.043] | Status  | Checking the format of the metadata in <COVAREP> computational sequence ...
[2021-11-08 14:11:54.043] | Warning | <COVAREP> computational sequence does not have all the required metadata ... continuing 
[2021-11-08 14:11:54.272] | Success | Computational sequence read from file cmumosi/CMU_MOSI_Opinion_Labels.csd ...
[2021-11-08 14:11:54.309] | Status  | Checking the integrity of the <Opinion Segment Labels> computational sequence ...
[2021-11-08 14:11:54.309] | Status  | Checking the format of the data in <Opinion Segment Labels> computational sequence ...


[2021-11-08 14:11:54.397] | Success | <Opinion Segment Labels> computational sequence data in correct format.
[2021-11-08 14:11:54.397] | Status  | Checking the format of the metadata in <Opinion Segment Labels> computational sequence ...
[2021-11-08 14:11:54.397] | Warning | <Opinion Segment Labels> computational sequence does not have all the required metadata ... continuing 
[2021-11-08 14:11:54.397] | Success | Dataset initialized successfully ... 


In [ ]:
#some random video from cmumosi_highlevel
some_video=list(cmumosi_highlevel["glove_vectors"].data.keys())[0]

In [ ]:
#The next line is not needed for standard datasets as they are all sorted based on intervals in computational sequence entries
cmumosi_highlevel.sort()

In [ ]:
#Aligning to the words to get word-level alignments
cmumosi_highlevel.align('glove_vectors',collapse_functions=[myavg])
cmumosi_highlevel.impute('glove_vectors')

[2021-11-08 06:21:57.913] | Status  | Unify was called ...
[2021-11-08 06:21:57.914] | Warning | c5xsKMxpXnc entry is not shared among all sequences, removing it ...
[2021-11-08 06:21:57.918] | Success | Unify completed ...
[2021-11-08 06:21:57.918] | Status  | Pre-alignment based on <glove_vectors> computational sequence started ...
[2021-11-08 06:21:58.171] | Status  | Pre-alignment done for <OpenSmile_emobase2010> ...
[2021-11-08 06:21:59.795] | Status  | Pre-alignment done for <FACET_4.2> ...
[2021-11-08 06:22:09.353] | Status  | Pre-alignment done for <OpenFace_1> ...
[2021-11-08 06:22:10.994] | Status  | Pre-alignment done for <FACET_4.1> ...
[2021-11-08 06:22:23.471] | Status  | Pre-alignment done for <OpenFace_2> ...
[2021-11-08 06:22:29.174] | Status  | Pre-alignment done for <COVAREP> ...
[2021-11-08 06:22:37.206] | Status  | Pre-alignment done for <b'OpenSMILE'> ...
[2021-11-08 06:22:37.365] | Status  | Alignment starting ...


Aligning zhpQhgha_KU:  99%|█████████▊| 666/676 [00:05<00:00, 127.33 Segments/s]


[2021-11-08 06:28:18.992] | Success | Alignment to <glove_vectors> complete.
[2021-11-08 06:28:18.992] | Status  | Replacing dataset content with aligned computational sequences
[2021-11-08 06:28:19.025] | Success | Initialized empty <glove_vectors> computational sequence.
[2021-11-08 06:28:19.026] | Status  | Checking the format of the data in <glove_vectors> computational sequence ...


[2021-11-08 06:28:19.252] | Success | <glove_vectors> computational sequence data in correct format.
[2021-11-08 06:28:19.252] | Status  | Checking the format of the metadata in <glove_vectors> computational sequence ...
[2021-11-08 06:28:19.252] | Warning | <glove_vectors> computational sequence does not have all the required metadata ... continuing 
[2021-11-08 06:28:19.252] | Success | Initialized empty <FACET_4.1> computational sequence.
[2021-11-08 06:28:19.252] | Status  | Checking the format of the data in <FACET_4.1> computational sequence ...


[2021-11-08 06:28:19.410] | Success | <FACET_4.1> computational sequence data in correct format.
[2021-11-08 06:28:19.410] | Status  | Checking the format of the metadata in <FACET_4.1> computational sequence ...
[2021-11-08 06:28:19.410] | Warning | <FACET_4.1> computational sequence does not have all the required metadata ... continuing 
[2021-11-08 06:28:19.410] | Success | Initialized empty <FACET_4.2> computational sequence.
[2021-11-08 06:28:19.410] | Status  | Checking the format of the data in <FACET_4.2> computational sequence ...


[2021-11-08 06:28:19.565] | Success | <FACET_4.2> computational sequence data in correct format.
[2021-11-08 06:28:19.565] | Status  | Checking the format of the metadata in <FACET_4.2> computational sequence ...
[2021-11-08 06:28:19.565] | Warning | <FACET_4.2> computational sequence does not have all the required metadata ... continuing 
[2021-11-08 06:28:19.565] | Success | Initialized empty <OpenSmile_emobase2010> computational sequence.
[2021-11-08 06:28:19.566] | Status  | Checking the format of the data in <OpenSmile_emobase2010> computational sequence ...


[2021-11-08 06:28:19.664] | Success | <OpenSmile_emobase2010> computational sequence data in correct format.
[2021-11-08 06:28:19.664] | Status  | Checking the format of the metadata in <OpenSmile_emobase2010> computational sequence ...
[2021-11-08 06:28:19.664] | Warning | <OpenSmile_emobase2010> computational sequence does not have all the required metadata ... continuing 
[2021-11-08 06:28:19.664] | Success | Initialized empty <b'OpenSMILE'> computational sequence.
[2021-11-08 06:28:19.672] | Status  | Checking the format of the data in <b'OpenSMILE'> computational sequence ...


[2021-11-08 06:28:19.819] | Success | <b'OpenSMILE'> computational sequence data in correct format.
[2021-11-08 06:28:19.820] | Status  | Checking the format of the metadata in <b'OpenSMILE'> computational sequence ...
[2021-11-08 06:28:19.820] | Warning | <b'OpenSMILE'> computational sequence does not have all the required metadata ... continuing 
[2021-11-08 06:28:19.820] | Success | Initialized empty <OpenFace_1> computational sequence.
[2021-11-08 06:28:19.820] | Status  | Checking the format of the data in <OpenFace_1> computational sequence ...


[2021-11-08 06:28:19.978] | Success | <OpenFace_1> computational sequence data in correct format.
[2021-11-08 06:28:19.978] | Status  | Checking the format of the metadata in <OpenFace_1> computational sequence ...
[2021-11-08 06:28:19.979] | Warning | <OpenFace_1> computational sequence does not have all the required metadata ... continuing 
[2021-11-08 06:28:19.979] | Success | Initialized empty <OpenFace_2> computational sequence.
[2021-11-08 06:28:19.979] | Status  | Checking the format of the data in <OpenFace_2> computational sequence ...


[2021-11-08 06:28:20.008] | Success | <OpenFace_2> computational sequence data in correct format.
[2021-11-08 06:28:20.008] | Status  | Checking the format of the metadata in <OpenFace_2> computational sequence ...
[2021-11-08 06:28:20.009] | Warning | <OpenFace_2> computational sequence does not have all the required metadata ... continuing 
[2021-11-08 06:28:20.009] | Success | Initialized empty <COVAREP> computational sequence.
[2021-11-08 06:28:20.009] | Status  | Checking the format of the data in <COVAREP> computational sequence ...


[2021-11-08 06:28:20.162] | Success | <COVAREP> computational sequence data in correct format.
[2021-11-08 06:28:20.162] | Status  | Checking the format of the metadata in <COVAREP> computational sequence ...
[2021-11-08 06:28:20.162] | Warning | <COVAREP> computational sequence does not have all the required metadata ... continuing 
[2021-11-08 06:28:20.201] | Status  | Imputation called with function: zeros ...


[2021-11-08 06:28:21.392] | Success | Imputation completed ...


In [ ]:
#get the intervals and features accompanying the 100th word in the some_video
some_video_100th_word=some_video+'[100]'
for compseq_name in list(cmumosi_highlevel.computational_sequences.keys()):
	compseq=cmumosi_highlevel[compseq_name]
	print (compseq_name)
	print (numpy.array(compseq.data[some_video_100th_word]["intervals"]).shape,numpy.array(compseq.data[some_video_100th_word]["features"]).shape)
	print ("-------")

glove_vectors
(1, 2) (1, 300)
-------
FACET_4.1
(1, 2) (1, 47)
-------
FACET_4.2
(1, 2) (1, 35)
-------
OpenSmile_emobase2010
(1, 2) (1, 1585)
-------
b'OpenSMILE'
(1, 2) (1, 384)
-------
OpenFace_1
(1, 2) (1, 430)
-------
OpenFace_2
(1, 2) (1, 713)
-------
COVAREP
(1, 2) (1, 74)
-------


k번째 비디오의 100번째 단어의 computational sequence를 관찰한 결과,

* Text - glove_vectors (feature: 300)

* Image - OpenFace_1 (430), FACET_4.2 (35), FACET_4.1 (47), COVAREP(74), OpenFace_2 (713)

* Audio - OpenSmile_emobase2010 (1585), b'OpenSMILE (384)

이를 통해 embedding feature 수 파악 가능.

*참고로 k번째 비디오는 glove_vector로 리스트업 하여 추출한 것.*

```python
#some random video from cmumosi_highlevel
some_video=list(cmumosi_highlevel["glove_vectors"].data.keys())[0]
```

![computational sequence](examples/compseq.png)

In [ ]:
cmumosi_highlevel.add_computational_sequences(mmdatasdk.cmu_mosi.labels,'cmumosi/')
cmumosi_highlevel.align('Opinion Segment Labels')
cmumosi_highlevel.hard_unify()

[2021-11-08 06:30:55.543] | Status  | Downloading from http://immortal.multicomp.cs.cmu.edu/CMU-MOSI/labels/CMU_MOSI_Opinion_Labels.csd to cmumosi/CMU_MOSI_Opinion_Labels.csd...


[2021-11-08 06:30:55.718] | Success | Download complete!
[2021-11-08 06:30:55.721] | Success | Computational sequence read from file cmumosi/CMU_MOSI_Opinion_Labels.csd ...


[2021-11-08 06:30:55.757] | Status  | Checking the integrity of the <Opinion Segment Labels> computational sequence ...
[2021-11-08 06:30:55.757] | Status  | Checking the format of the data in <Opinion Segment Labels> computational sequence ...


[2021-11-08 06:30:55.842] | Success | <Opinion Segment Labels> computational sequence data in correct format.
[2021-11-08 06:30:55.842] | Status  | Checking the format of the metadata in <Opinion Segment Labels> computational sequence ...
[2021-11-08 06:30:55.842] | Warning | <Opinion Segment Labels> computational sequence does not have all the required metadata ... continuing 
[2021-11-08 06:30:55.842] | Status  | Unify was called ...
[2021-11-08 06:30:56.214] | Warning | c5xsKMxpXnc entry is not shared among all sequences, removing it ...
[2021-11-08 06:30:56.329] | Success | Unify completed ...
[2021-11-08 06:30:56.337] | Status  | Pre-alignment based on <Opinion Segment Labels> computational sequence started ...
[2021-11-08 06:30:56.776] | Status  | Pre-alignment done for <OpenSmile_emobase2010> ...
[2021-11-08 06:30:56.935] | Status  | Pre-alignment done for <FACET_4.2> ...
[2021-11-08 06:30:57.166] | Status  | Pre-alignment done for <OpenFace_1> ...
[2021-11-08 06:30:57.340] | St

Aligning 03bSnISJMiM:   0%|          | 0/13 [00:00<?, ? Segments/s]
                                                                   
Aligning 1DmNV9C1hbY:   0%|          | 0/14 [00:00<?, ? Segments/s]
                                                                   
Aligning 2WGyTLYerpo:  52%|█████▏    | 33/63 [00:00<00:00, 328.08 Segments/s]
                                                                             
Aligning 6Egk_28TtTM:   0%|          | 0/12 [00:00<?, ? Segments/s]
                                                                   
Aligning 9c67fiY0wGQ:   0%|          | 0/12 [00:00<?, ? Segments/s]
                                                                   
Aligning G-xst2euQUc:   0%|          | 0/21 [00:00<?, ? Segments/s]
                                                                   
Aligning Iu2PFX3z_1s:   0%|          | 0/16 [00:00<?, ? Segments/s]
                                                                   
Aligning Njd1F0vZSm4:   0%| 

[2021-11-08 06:31:09.442] | Success | Alignment to <Opinion Segment Labels> complete.
[2021-11-08 06:31:09.442] | Status  | Replacing dataset content with aligned computational sequences
[2021-11-08 06:31:10.114] | Success | Initialized empty <glove_vectors> computational sequence.
[2021-11-08 06:31:10.119] | Status  | Checking the format of the data in <glove_vectors> computational sequence ...


[2021-11-08 06:31:10.139] | Success | <glove_vectors> computational sequence data in correct format.
[2021-11-08 06:31:10.139] | Status  | Checking the format of the metadata in <glove_vectors> computational sequence ...
[2021-11-08 06:31:10.139] | Warning | <glove_vectors> computational sequence does not have all the required metadata ... continuing 
[2021-11-08 06:31:10.139] | Success | Initialized empty <FACET_4.1> computational sequence.
[2021-11-08 06:31:10.139] | Status  | Checking the format of the data in <FACET_4.1> computational sequence ...


[2021-11-08 06:31:10.157] | Success | <FACET_4.1> computational sequence data in correct format.
[2021-11-08 06:31:10.157] | Status  | Checking the format of the metadata in <FACET_4.1> computational sequence ...
[2021-11-08 06:31:10.157] | Warning | <FACET_4.1> computational sequence does not have all the required metadata ... continuing 
[2021-11-08 06:31:10.157] | Success | Initialized empty <FACET_4.2> computational sequence.
[2021-11-08 06:31:10.157] | Status  | Checking the format of the data in <FACET_4.2> computational sequence ...


[2021-11-08 06:31:10.175] | Success | <FACET_4.2> computational sequence data in correct format.
[2021-11-08 06:31:10.176] | Status  | Checking the format of the metadata in <FACET_4.2> computational sequence ...
[2021-11-08 06:31:10.176] | Warning | <FACET_4.2> computational sequence does not have all the required metadata ... continuing 
[2021-11-08 06:31:10.176] | Success | Initialized empty <OpenSmile_emobase2010> computational sequence.
[2021-11-08 06:31:10.176] | Status  | Checking the format of the data in <OpenSmile_emobase2010> computational sequence ...


[2021-11-08 06:31:10.202] | Success | <OpenSmile_emobase2010> computational sequence data in correct format.
[2021-11-08 06:31:10.202] | Status  | Checking the format of the metadata in <OpenSmile_emobase2010> computational sequence ...
[2021-11-08 06:31:10.202] | Warning | <OpenSmile_emobase2010> computational sequence does not have all the required metadata ... continuing 
[2021-11-08 06:31:10.202] | Success | Initialized empty <b'OpenSMILE'> computational sequence.
[2021-11-08 06:31:10.202] | Status  | Checking the format of the data in <b'OpenSMILE'> computational sequence ...


[2021-11-08 06:31:10.221] | Success | <b'OpenSMILE'> computational sequence data in correct format.
[2021-11-08 06:31:10.221] | Status  | Checking the format of the metadata in <b'OpenSMILE'> computational sequence ...
[2021-11-08 06:31:10.221] | Warning | <b'OpenSMILE'> computational sequence does not have all the required metadata ... continuing 
[2021-11-08 06:31:10.221] | Success | Initialized empty <OpenFace_1> computational sequence.
[2021-11-08 06:31:10.221] | Status  | Checking the format of the data in <OpenFace_1> computational sequence ...


[2021-11-08 06:31:10.245] | Success | <OpenFace_1> computational sequence data in correct format.
[2021-11-08 06:31:10.245] | Status  | Checking the format of the metadata in <OpenFace_1> computational sequence ...
[2021-11-08 06:31:10.245] | Warning | <OpenFace_1> computational sequence does not have all the required metadata ... continuing 
[2021-11-08 06:31:10.245] | Success | Initialized empty <OpenFace_2> computational sequence.
[2021-11-08 06:31:10.245] | Status  | Checking the format of the data in <OpenFace_2> computational sequence ...


[2021-11-08 06:31:10.268] | Success | <OpenFace_2> computational sequence data in correct format.
[2021-11-08 06:31:10.268] | Status  | Checking the format of the metadata in <OpenFace_2> computational sequence ...
[2021-11-08 06:31:10.269] | Warning | <OpenFace_2> computational sequence does not have all the required metadata ... continuing 
[2021-11-08 06:31:10.269] | Success | Initialized empty <COVAREP> computational sequence.
[2021-11-08 06:31:10.269] | Status  | Checking the format of the data in <COVAREP> computational sequence ...


[2021-11-08 06:31:10.289] | Success | <COVAREP> computational sequence data in correct format.
[2021-11-08 06:31:10.289] | Status  | Checking the format of the metadata in <COVAREP> computational sequence ...
[2021-11-08 06:31:10.289] | Warning | <COVAREP> computational sequence does not have all the required metadata ... continuing 
[2021-11-08 06:31:10.290] | Success | Initialized empty <Opinion Segment Labels> computational sequence.
[2021-11-08 06:31:10.290] | Status  | Checking the format of the data in <Opinion Segment Labels> computational sequence ...


[2021-11-08 06:31:10.307] | Success | <Opinion Segment Labels> computational sequence data in correct format.
[2021-11-08 06:31:10.307] | Status  | Checking the format of the metadata in <Opinion Segment Labels> computational sequence ...
[2021-11-08 06:31:10.307] | Warning | <Opinion Segment Labels> computational sequence does not have all the required metadata ... continuing 
[2021-11-08 06:31:10.363] | Status  | Hard unify was called ...
[2021-11-08 06:31:10.372] | Success | Hard unify completed ...


In [ ]:
#get the intervals and features accompanying the 2nd in some_video
some_video_2nd_segment=some_video+'[2]'
for compseq_name in list(cmumosi_highlevel.computational_sequences.keys()):
	compseq=cmumosi_highlevel[compseq_name]
	print (compseq_name)
	print (numpy.array(compseq.data[some_video_2nd_segment]["intervals"]).shape,numpy.array(compseq.data[some_video_2nd_segment]["features"]).shape)
	print ("-------")

glove_vectors
(10, 2) (10, 300)
-------
FACET_4.1
(10, 2) (10, 47)
-------
FACET_4.2
(10, 2) (10, 35)
-------
OpenSmile_emobase2010
(10, 2) (10, 1585)
-------
b'OpenSMILE'
(10, 2) (10, 384)
-------
OpenFace_1
(10, 2) (10, 430)
-------
OpenFace_2
(10, 2) (10, 713)
-------
COVAREP
(10, 2) (10, 74)
-------
Opinion Segment Labels
(1, 2) (1, 1)
-------


In [ ]:
#Deploying the files to the disk and reading them again - Building machine learning models start right after this. No need to do alignment multiple times since aligned files can be deployed and used again.
# deploy_files={x:x for x in cmumosi_highlevel.computational_sequences.keys()}
# cmumosi_highlevel.deploy("./deployed",deploy_files)

# #Reading the dumped file can be as easy as just calling the mmdataset on the deployed folder
# aligned_cmumosi_highlevel=mmdatasdk.mmdataset('./deployed')

#Now let's get the tensor ready for ML - right here we just get everything into ML ready tensors. But you should split the aligned_cmumosi_highlevel based on the standard CMU MOSI folds
#get the standard folds using mmsdk.mmdatasdk.cmu_mosi.standard_folds.standard_x_fold for x={train,test,valid}


In [ ]:
tensors=cmumosi_highlevel.get_tensors(seq_len=25,non_sequences=["Opinion Segment Labels"],direction=False,folds=[mmdatasdk.cmu_mosi.standard_folds.standard_train_fold,mmdatasdk.cmu_mosi.standard_folds.standard_valid_fold,mmdatasdk.cmu_mosi.standard_folds.standard_test_fold])

fold_names=["train","valid","test"]

for i in range(3):
	for csd in list(cmumosi_highlevel.keys()):
		print ("Shape of the %s computational sequence for %s fold is %s"%(csd,fold_names[i],tensors[i][csd].shape))


[2021-11-08 06:32:36.956] | Status  | Hard unify was called ...
[2021-11-08 06:32:36.964] | Success | Hard unify completed ...
Shape of the glove_vectors computational sequence for train fold is (1283, 25, 300)
Shape of the FACET_4.1 computational sequence for train fold is (1283, 25, 47)
Shape of the FACET_4.2 computational sequence for train fold is (1283, 25, 35)
Shape of the OpenSmile_emobase2010 computational sequence for train fold is (1283, 25, 1585)
Shape of the b'OpenSMILE' computational sequence for train fold is (1283, 25, 384)
Shape of the OpenFace_1 computational sequence for train fold is (1283, 25, 430)
Shape of the OpenFace_2 computational sequence for train fold is (1283, 25, 713)
Shape of the COVAREP computational sequence for train fold is (1283, 25, 74)
Shape of the Opinion Segment Labels computational sequence for train fold is (1283, 1, 1)
Shape of the glove_vectors computational sequence for valid fold is (214, 25, 300)
Shape of the FACET_4.1 computational sequen

In [ ]:
print(train[0])

NameError: ignored